File: app/src/main/java/mastermind/services/SettingsSaveStateService.java
package mastermind.services;

import android.content.Context;
import android.content.SharedPreferences;

public class SettingsSaveStateService {
    private static final String FILE_NAME = "mastermind.preferences";
    private static final String SAVESTATE_KEY = "save";
    private static final String LANGUAGE = "language";
    private static final String REPEAT_COLORS = "repeatable";
    private static final String EMPTY = "none";
    private static SharedPreferences preferences;

    public static void initialize(Context context) {
        preferences = context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);
    }

    public static boolean hasSavedGame() {
        String savestate = getSavedGame();
        return !savestate.equals(EMPTY);
    }

    public static String getSavedGame() {
        return preferences.getString(SAVESTATE_KEY, EMPTY);
    }

    public static void saveGame(String gamestate) {
        preferences.edit().putString(SAVESTATE_KEY, gamestate).apply();
    }
    public static boolean hasLanguage() {
        String language = getLanguage();
        return !language.equals(EMPTY);
    }

    public static String getLanguage() {
        return preferences.getString(LANGUAGE, EMPTY);
    }

    public static void saveLanguage(String language) {
        preferences.edit().putString(LANGUAGE, language).apply();
    }
    public static boolean hasRepeatColors() {
        String repeatable = getRepeatColors();
        return !repeatable.equals(EMPTY);
    }

    public static String getRepeatColors() {
        return preferences.getString(REPEAT_COLORS, EMPTY);
    }

    public static void saveRepeatColors(String repeatable) {
        preferences.edit().putString(REPEAT_COLORS, repeatable).apply();
    }


}

File: app/src/main/kotlin/mastermind/services/SettingsSaveStateService.kt
package mastermind.services

import android.content.Context
import android.content.SharedPreferences
import mastermind.services.SettingsSaveStateService

object SettingsSaveStateService {
    private const val FILE_NAME = "mastermind.preferences"
    private const val SAVESTATE_KEY = "save"
    private const val LANGUAGE = "language"
    private const val REPEAT_COLORS = "repeatable"
    private const val EMPTY = "none"
    private const val DARKMODE = "darkmode"
    private var preferences: SharedPreferences? = null
    @JvmStatic
    fun initialize(context: Context) {
        preferences = context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
    }

    fun hasSavedGame(): Boolean {
        val savestate = savedGame
        return savestate != EMPTY
    }

    val savedGame: String?
        get() = preferences!!.getString(SAVESTATE_KEY, EMPTY)

    fun saveGame(gamestate: String?) {
        preferences!!.edit().putString(SAVESTATE_KEY, gamestate).apply()
    }

    fun hasLanguage(): Boolean {
        val language = language
        return language != EMPTY
    }

    val language: String?
        get() = preferences!!.getString(LANGUAGE, EMPTY)

    fun saveLanguage(language: String?) {
        preferences!!.edit().putString(LANGUAGE, language).apply()
    }

    fun hasRepeatColors(): Boolean {
        val repeatable = repeatColors
        return repeatable != EMPTY
    }

    val repeatColors: String?
        get() = preferences!!.getString(REPEAT_COLORS, EMPTY)

    fun saveRepeatColors(repeatable: String?) {
        preferences!!.edit().putString(REPEAT_COLORS, repeatable).apply()
    }

    fun saveGame(savestring: String?, repeatable: String?) {
        saveGame(savestring)
        saveRepeatColors(repeatable)
    }

    val isDarkmodeOn: Boolean
        get() = "true" == preferences!!.getString(DARKMODE, EMPTY)

    fun saveDarkmode(isDarkmode: Boolean) {
        preferences!!.edit().putString(DARKMODE, if (isDarkmode) "true" else "false").apply()
    }
}